engine/src/main/java/org/hibernate/validator/internal/properties/DefaultGetterPropertySelectionStrategy.java

```diff
@@ -0,0 +1,99 @@
+import java.util.Set;
+
+import org.hibernate.validator.internal.util.CollectionHelper;
+import org.hibernate.validator.internal.util.Contracts;
+import org.hibernate.validator.internal.util.StringHelper;
+import org.hibernate.validator.internal.util.logging.Log;
+import org.hibernate.validator.internal.util.logging.LoggerFactory;
+import org.hibernate.validator.spi.properties.ConstrainableExecutable;
+import org.hibernate.validator.spi.properties.GetterPropertySelectionStrategy;
+
+/**
+ * @author Marko Bekhta
+ */
+public class DefaultGetterPropertySelectionStrategy implements GetterPropertySelectionStrategy {
+
+	private static final Log LOG = LoggerFactory.make( MethodHandles.lookup() );
+
+	private static final String GETTER_PREFIX_GET = "get";
+	private static final String GETTER_PREFIX_IS = "is";
+	private static final String GETTER_PREFIX_HAS = "has";
+	private static final String[] GETTER_PREFIXES = {
+			GETTER_PREFIX_GET,
+			GETTER_PREFIX_IS,
+			GETTER_PREFIX_HAS
+	};
+
+	/**
+	 * Checks whether the given executable is a valid JavaBeanHelper getter method, which
+	 * is the case if
+	 * <ul>
+	 * <li>its name starts with "get" and it has a return type but no parameter or</li>
+	 * <li>its name starts with "is", it has no parameter and is returning
+	 * {@code boolean} or</li>
+	 * <li>its name starts with "has", it has no parameter and is returning
+	 * {@code boolean} (HV-specific, not mandated by the JavaBeans spec).</li>
+	 * </ul>
+	 *
+	 * @param executable The executable of interest.
+	 *
+	 * @return {@code true}, if the given executable is a JavaBeanHelper getter method,
+	 * {@code false} otherwise.
+	 */
+	@Override
+	public boolean isGetter(ConstrainableExecutable executable) {
+		if ( executable.getParameterTypes().length != 0 ) {
+			return false;
+		}
+
+		String methodName = executable.getName();
+
+		//<PropertyType> get<PropertyName>()
+		if ( methodName.startsWith( GETTER_PREFIX_GET ) && executable.getReturnType() != void.class ) {
+			return true;
+		}
+		//boolean is<PropertyName>()
+		else if ( methodName.startsWith( GETTER_PREFIX_IS ) && executable.getReturnType() == boolean.class ) {
+			return true;
+		}
+		//boolean has<PropertyName>()
+		else if ( methodName.startsWith( GETTER_PREFIX_HAS ) && executable.getReturnType() == boolean.class ) {
+			return true;
+		}
+
+		return false;
+	}
+
+	@Override
+	public String getPropertyName(ConstrainableExecutable executable) {
+		Contracts.assertNotNull( executable, "executable cannot be null" );
+
+		String methodName = executable.getName();
+		for ( String prefix : GETTER_PREFIXES ) {
+			if ( methodName.startsWith( prefix ) ) {
+				return StringHelper.decapitalize( methodName.substring( prefix.length() ) );
+			}
+		}
+		throw LOG.getIllegalArgumentException( "Property name cannot be constructed for a given method " + executable.getName() );
+	}
+
+	@Override
+	public Set<String> getPossibleMethodNamesForGetterProperty(String propertyName) {
```

Comment:
Maybe `getGetterNameCandidates()`?

If so, let's make the core of the method consistent with the new name (e.g. variable names).